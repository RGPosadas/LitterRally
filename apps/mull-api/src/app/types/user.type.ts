import { Field, ID, ObjectType } from '@nestjs/graphql';

@ObjectType('User')
export class UserType {
  @Field(/* istanbul ignore next */ () => ID)
  id: number;

  @Field()
  password: string;

  @Field()
  email: string;

  @Field()
  timezone: string;

  @Field()
  name: string;

  @Field()
  age: string;

  @Field()
  description: string;

  @Field(/* istanbul ignore next */ () => [UserType])
  friends: UserType[];

  // TODO: Implement modules, services, types for Entities below

  //   @Field(() => [EventType])
  //   events: EventType[];

  //   @Field(() => [PostReactionType])
  //   postReactions: PostReactionType[];

  //   @Field(() => MediaType)
  //   avatar: MediaType;
}
