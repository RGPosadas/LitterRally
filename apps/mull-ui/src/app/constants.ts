export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
