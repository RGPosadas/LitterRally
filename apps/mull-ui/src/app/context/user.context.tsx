import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
