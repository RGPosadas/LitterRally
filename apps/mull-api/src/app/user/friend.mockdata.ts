import { RegistrationMethod } from '@mull/types';
import { mockAllDirectMessageChannels } from '../channel/channel.mockdata';
import { Friend } from '../entities';
import { mockAllPosts } from '../post/post.mockdata';

// Friends of mockAllUsers[0]
export const mockAllFriends: Friend[] = [
  {
    id: 7,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Cristian',
    dob: new Date(),
    description: "There's a first for everything",
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: mockAllPosts[3],
    directMessageChannel: mockAllDirectMessageChannels[0],
  },
  {
    id: 12,
    password: 'abc123',
    email: 'tim@green.co',
    timezone: '',
    name: 'Tim',
    dob: new Date(),
    description: '',
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: null,
    directMessageChannel: null,
  },
];
