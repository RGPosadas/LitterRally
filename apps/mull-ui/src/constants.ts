import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
