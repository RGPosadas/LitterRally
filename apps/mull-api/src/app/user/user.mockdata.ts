import { RegistrationMethod } from '@mull/types';
import { User } from '../entities';
import { CreateUserInput, UpdateUserInput } from './inputs/user.input';

export const mockPartialUser: CreateUserInput | UpdateUserInput = {
  password: 'password',
  email: 'mock@mock.com',
  dob: new Date(),
  name: 'Mock McMockson',
  registrationMethod: RegistrationMethod.LOCAL,
};

export const mockNewPartialUser: CreateUserInput = {
  password: 'password',
  email: 'mock2@mock.com',
  dob: new Date(),
  name: 'Mock McMockson Jr.',
  registrationMethod: RegistrationMethod.LOCAL,
};

export const mockAllUsers: User[] = [
  {
    id: 1,
    password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
    email: 'mock@mock.com',
    timezone: '',
    name: 'Bob',
    dob: new Date(),
    description: 'I am very good programmer',
    friends: [
      {
        id: 7,
        password: 'password',
        email: 'me@me.com',
        timezone: '',
        name: 'Cristian',
        dob: new Date(),
        description: "There's a first for everything",
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
    ],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
  },
  {
    id: 7,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Cristian',
    dob: new Date(),
    description: "There's a first for everything",
    friends: [
      {
        id: 1,
        password: 'abc123',
        email: 'gr@ph.ql',
        timezone: '',
        name: 'Bob',
        dob: new Date(),
        description: 'I am very good programmer',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        tokenVersion: 0,
        registrationMethod: RegistrationMethod.LOCAL,
      },
    ],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.GOOGLE,
  },
  {
    id: 3,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Jose',
    dob: new Date(),
    description: 'What goes up, comes down',
    friends: [
      {
        id: 1,
        password: 'abc123',
        email: 'gr@ph.ql',
        timezone: '',
        name: 'Bob',
        dob: new Date(),
        description: 'I am very good programmer',
        friends: [],
        registrationMethod: RegistrationMethod.LOCAL,
        tokenVersion: 0,
      },
      {
        id: 12,
        password: 'abc123',
        email: 'tim@green.co',
        timezone: '',
        name: 'Tim',
        dob: new Date(),
        description: '',
        friends: [],
        registrationMethod: RegistrationMethod.LOCAL,
        tokenVersion: 0,
      },
    ],
    registrationMethod: RegistrationMethod.GOOGLE,
    tokenVersion: 0,
  },
];
