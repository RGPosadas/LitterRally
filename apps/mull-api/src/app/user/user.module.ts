import { Module } from '@nestjs/common';
import { EntitiesModule } from '../entities';
import { EventService } from '../event/event.service';
import { MediaService } from '../media/media.service';
import { UserResolver } from './user.resolver';
import { UserService } from './user.service';

@Module({
  imports: [EntitiesModule],
  providers: [UserResolver, UserService, EventService, MediaService],
  exports: [UserService],
})
export class UserModule {}
