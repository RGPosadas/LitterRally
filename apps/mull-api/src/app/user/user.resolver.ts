import { RegistrationMethod } from '@mull/types';
import { UnauthorizedException } from '@nestjs/common';
import { Args, Int, Mutation, Parent, Query, ResolveField, Resolver } from '@nestjs/graphql';
import { genSalt, hash } from 'bcrypt';
import { User } from '../entities';
import { CreateUserInput, UpdateUserInput } from './inputs/user.input';
import { UserService } from './user.service';

@Resolver(/* istanbul ignore next */ () => User)
export class UserResolver {
  constructor(private readonly userService: UserService) {}

  @Query(/* istanbul ignore next */ () => [User])
  async users() {
    return this.userService.findAll();
  }

  @Query(/* istanbul ignore next */ () => User)
  async user(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
    return this.userService.findOne(id);
  }

  @ResolveField(/* istanbul ignore next */ () => [User])
  async friends(@Parent() user: User) {
    const { id } = user;
    return this.userService.findAllFriends(id);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async createUser(@Args('createUserInput') createUserInput: CreateUserInput) {
    if (createUserInput.registrationMethod === RegistrationMethod.LOCAL) {
      const salt = await genSalt(10);
      const hashed = await hash(createUserInput.password, salt);
      createUserInput.password = hashed;
    }
    const existingUser = await this.userService.findUnique(
      createUserInput.email,
      createUserInput.registrationMethod
    );
    if (existingUser.length > 0)
      throw new UnauthorizedException('User with this email already exists.');
    return this.userService.create(createUserInput);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async updateUser(@Args('updateUserInput') updateUserInput: UpdateUserInput) {
    return this.userService.updateUser(updateUserInput);
  }

  @Mutation(/* istanbul ignore next */ () => User)
  async deleteUser(@Args('id', { type: /* istanbul ignore next */ () => Int }) id: number) {
    return this.userService.delete(id);
  }
}
