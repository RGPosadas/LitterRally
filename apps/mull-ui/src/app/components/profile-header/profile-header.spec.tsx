import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { FriendStatusButton } from '@mull/types';
import '@testing-library/jest-dom/extend-expect';
import { act, render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer, { act as actRenderer } from 'react-test-renderer';
import { ROUTES } from '../../../constants';
import {
  RegistrationMethod,
  RelationshipType,
  User,
  UserRelationshipDocument,
  UserRelationshipQuery,
} from '../../../generated/graphql';
import { UserProvider } from '../../context/user.context';
import ProfileHeader from './profile-header';

describe('ProfileHeader', () => {
  const renderHelper = (history, mocks: MockedResponse[], isCurrentUser: boolean, user: User) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <Router history={history}>
          <MockedProvider mocks={mocks} addTypename={false}>
            <ProfileHeader
              isCurrentUser={isCurrentUser}
              portfolioCount={2}
              friendCount={0}
              hostingCount={3}
              user={user}
            />
          </MockedProvider>
        </Router>
      </UserProvider>
    );
  };

  const mockUser: User = {
    id: 4,
    name: 'Candice Runolfsdottir',
    description:
      "Hey, it's Candice! Thanks for following my profile~ I am really passionate about hte environment, so feel free to add me and let us go around helping the environment together!",
    joinDate: '2020-04-28T06:03:28.000Z',
    avatar: null,
    timezone: '',
    email: 'candice@candice.com',
    friends: [],
    registrationMethod: RegistrationMethod.Local,
  };

  it('should render successfully', async () => {
    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    const { baseElement } = render(renderHelper(history, null, false, mockUser));
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserRelationshipDocument,
          variables: { userIdB: mockUser.id },
        },
        result: {
          data: { getUserRelationship: RelationshipType.None } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await actRenderer(async () => {
      const tree = renderer.create(renderHelper(history, mocks, false, mockUser));
      await new Promise((resolve) => setTimeout(resolve, 200));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  it('should render Friends button successfully', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserRelationshipDocument,
          variables: { userIdB: mockUser.id },
        },
        result: {
          data: { getUserRelationship: RelationshipType.Friends } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await act(async () => {
      const utils = render(renderHelper(history, mocks, false, mockUser));
      await new Promise((resolve) => setTimeout(resolve, 200));
      const friendStatusButton = utils.container.querySelector('button[data-testid="mull-button"]');
      expect(friendStatusButton).toHaveTextContent(FriendStatusButton.FRIENDS);
    });
  });

  it('should render Pending button successfully', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserRelationshipDocument,
          variables: { userIdB: mockUser.id },
        },
        result: {
          data: { getUserRelationship: RelationshipType.PendingRequest } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await act(async () => {
      const utils = render(renderHelper(history, mocks, false, mockUser));
      await new Promise((resolve) => setTimeout(resolve, 200));
      const friendStatusButton = utils.container.querySelector('button[data-testid="mull-button"]');
      expect(friendStatusButton).toHaveTextContent(FriendStatusButton.PENDING);
    });
  });

  it('should render Add Friend button succesfully when RelationshipType is NONE', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserRelationshipDocument,
          variables: { userIdB: mockUser.id },
        },
        result: {
          data: { getUserRelationship: RelationshipType.None } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await act(async () => {
      const utils = render(renderHelper(history, mocks, false, mockUser));
      await new Promise((resolve) => setTimeout(resolve, 200));
      const friendStatusButton = utils.container.querySelector('button[data-testid="mull-button"]');
      expect(friendStatusButton).toHaveTextContent(FriendStatusButton.ADD_FRIEND);
    });
  });

  it('should render Add Friend button succesfully when RelationshipType is ADDED_ME', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserRelationshipDocument,
          variables: { userIdB: mockUser.id },
        },
        result: {
          data: { getUserRelationship: RelationshipType.AddedMe } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await act(async () => {
      const utils = render(renderHelper(history, mocks, false, mockUser));
      await new Promise((resolve) => setTimeout(resolve, 200));
      const friendStatusButton = utils.container.querySelector('button[data-testid="mull-button"]');
      expect(friendStatusButton).toHaveTextContent(FriendStatusButton.ADD_FRIEND);
    });
  });
});
