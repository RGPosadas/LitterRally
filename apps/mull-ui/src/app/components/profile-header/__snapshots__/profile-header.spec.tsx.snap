// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader should match snapshot 1`] = `
<div
  className="profile-header-container"
>
  <div
    className="user-name-container"
  >
    <h1>
      
    </h1>
    <button
      className="share-button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-share-alt fa-w-14 "
        data-icon="share-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="profile-header"
  >
    <img
      alt="user"
      className="user-profile-picture"
      src=""
    />
    <div
      className="profile-side-container for-current-user"
    >
      <button
        className="profile-stats"
      >
        0
        <br />
        Portfolio
      </button>
      <button
        className="profile-stats"
      >
        0
        <br />
        Friends
      </button>
      <button
        className="profile-stats"
      >
        0
        <br />
        Hosting
      </button>
    </div>
  </div>
  <div
    className="user-description-container"
  >
    <p>
      
    </p>
  </div>
</div>
`;
